### PR TITLE
feat(curriculum): add password generator project

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

Adding a new project the Python curriculum.

### CODE REVIEW

**Comment by zairahira** [comment]
I ran the code, works perfectly fine.
I do not have any suggestions as the code is already readable and elegant. :rocket:

File: scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/password_generator.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    while True:
+        password = ''    
+        # Generate password
+        for i in range(length):
+            password += secrets.choice(all_characters)
+
+        # Check constraints        
+        constraints = [
+            (nums, re.findall('[\d]', password)),
+            (special_chars, re.findall('[\W]', password)),
```
[the remark this answers is not in the document]
Comment (Dario-DC) [reply scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/password_generator.py:23]: Just an oversight, I've started writing the [A-Z] character class and then I kept the same format without thinking.

File: scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/password_generator.py
Comment (gikf) [inline scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/password_generator.py:1]: :+1:

Last thing (nitpick) that occurred to me now - another style convention - using 2 blank lines before and after function definitions.

Comment (Dario-DC) [reply scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/password_generator.py:1]: I'm not sure we are following that convention for the other projects 🤔

Comment (gikf) [reply scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/password_generator.py:1]: One step at a time :slightly_smiling_face:

**Pull request merged**